core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -30,7 +30,7 @@ import org.apache.kafka.common.utils.Utils
 abstract class AbstractFetcherManager(protected val name: String, clientId: String, numFetchers: Int = 1)
   extends Logging with KafkaMetricsGroup {
   // map of (source broker_id, fetcher_id per source broker) => fetcher
-  private val fetcherThreadMap = new mutable.HashMap[BrokerIdAndFetcherId, AbstractFetcherThread]
+  val fetcherThreadMap = new mutable.HashMap[BrokerIdAndFetcherId, AbstractFetcherThread]
```

Comment:
We should try to avoid making mutable collections public. It makes it harder to reason about the code since it could be mutated by anything. If we need to expose some functionality for testing, then:

1. Keep it as private as possible. You can do `private[server]` for example.
2. We should add a comment stating that it's visible for testing
3. We should consider exposing a method with the functionality needed by the test (this is typically much more constrained).

Reply:
@ijuma Sure. This makes sense. I will add comment and make it `private[server]` in the pull request 3718.

Reply:
I will update the test in pull request 3718 to move LogDirFailureTest to kafka.server package and make this variable private[server].